fml/platform/posix/file_posix.cc

```diff
@@ -210,4 +211,20 @@ bool WriteAtomically(const fml::UniqueFD& base_directory,
                     base_directory.get(), file_name) == 0;
 }
 
+std::vector<std::string> ListFiles(const fml::UniqueFD& directory) {
+  std::vector<std::string> files;
+  DIR* dir = ::fdopendir(directory.get());
+  if (dir == nullptr) {
+    FML_LOG(ERROR) << "Can't open the directory.";
+    return files;
+  }
+  while (dirent* ent = readdir(dir)) {
+    std::string name = ent->d_name;
+    if (name != "." && name != "..") {
+      files.push_back(std::move(name));
```

Comment:
Optional suggestion: Consider using a visitor pattern here. For example, `size_t ListFiles(const fml::UniqueFD& directory, bool(const std::string& file) file_callback)` should work and is way more flexible IMO. With this in place, you can also make this call recursive if the directory entry is itself a directory. The `bool` return from the visitor callback can be an indication to the caller about whether further traversal is necessary.